File: Ruby/harmlessRansomNote.rb
def harmlessRansomNote(magazine_text, note_text )
  magazine_frequency = invoke_frequency(magazine_text)
  noteArr = note_text.split(' ')
  magazine_frequency.map do |key, value|
  end
end

def invoke_frequency(text)
  words = text.split(' ')
  frequency = {}
  words_list.map do |word|
    frequency_hash[word] = 0 if frequency_hash[word].nil?
    frequency_hash[word] += 1
  end
  frequency_hash
end

harmlessRansomNote(article, text)

File: JavaScript/harmlessRansomNote.js
function harmlessRansomNote(magazineText, noteText) {
  const noteArray = noteText.split(' ')
  const magazineArray = magazineText.split(' ')
  
  const magazineFrequency = {}
  let possibleNote = true

  magazineArray.map(word => {
    if (!word in magazineFrequency) magazineFrequency[word] = 0
    magazineFrequency[word]++
  })

  noteArray.map(word => {
    if (word in magazineFrequency) {
      magazineFrequency[word]--
      if (magazineFrequency[word] < 0) possibleNote = false
    } else {
      possibleNote = false
    }
  })
}
